src/registry.ts

```diff
@@ -88,17 +68,12 @@ export async function updateRegistry(
       data: {
         publickey: {
           algorithm: "ed25519",
-          key: Array.from(user.publicKey),
+          key: Array.from(publickey),
         },
-        fileid: {
-          version: fileID.version,
-          applicationid: fileID.applicationID,
-          filetype: fileID.fileType,
-          filename: fileID.filename,
-        },
-        revision: srv.value.revision,
-        data: Array.from(Uint8Array.from(Buffer.from(srv.value.data))),
-        signature: Array.from(Uint8Array.from(srv.signature)),
+        datakey,
+        revision: entry.revision,
+        data: Array.from(Uint8Array.from(Buffer.from(entry.data))),
```

Comment:
Are you sure these 3 are needed? Wouldn't this suffice? Or even maybe just `Array.from` ? (not tested it, just asking because it looks pretty entangled)
```suggestion
        data: Array.from(Buffer.from(entry.data)),
```